benchmarks/transport/mpi_p2p_bi_avail_mt.cpp

```diff
@@ -156,5 +207,5 @@ int main(int argc, char *argv[])
     } 
 
     MPI_Barrier(MPI_COMM_WORLD);
-    MPI_Finalize();
+    // MPI_Finalize(); // segfault due to unfinished requests
```

Comment:
Don't we have any mean to clean this up for the benchmark?

Reply:
@mbianco That is a problem. We cannot cancel send requests, so in our ghex benchmarks I deal with this by a tail-loop, which waits for all send requests to complete, and they by canceling all recv requests. We could probably do this here as well, but it is a lot of code - actually more than the test loop itself. It was much simpler to simply not call `MPI_Finalize()` - so far I don't see any side-effects, except for maybe some warnings, which can be turned off.

But if you think it's worth it, we can probably make a 'correct' test.

Reply:
It is not clear to me what value a benchmark has if it does not represents code that can be written safely by a user...

Reply:
The benchmark loop is important and representative of user application performance. The tail loops and cleanup loops are not part of the benchmark. Their only purpose is to exit the program.

In actual user application the users either send/recv _exactly_ the needed number of messages, all of which are always matched/completed (halo exchange in a static parallel environment) - and hence never end up in this problem, or they have to implement something similar to the tail-loops in order to handle rank migration etc (DISPATCH). If you pre-post a recv message, which will never arrive, you have to do something with it. This use-case is not frequent, application-specific, and we do not measure the time of that.